alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -185,4 +188,19 @@ bool BasicBehaviour::getParameter(const std::string& key, std::string& valueOut)
     }
 }
 
+void BasicBehaviour::startTrace()
+{
+    RunningPlan* parent = _context.load()->getParent();
+    if (parent == nullptr) {
+        // a behaviour always has a plan as its parent
+        return;
+    }
+
+    if (auto parentPlan = parent->getBasicPlan()) {
+        _trace = std::make_unique<IAlicaTrace>("Behaviour", parentPlan->getTraceContext());
```

Comment:
use the trace factory to create the trace. The name of the trace should be the behaviour name